### DISCUSSION TITLE
Custom Whitelist Arsenal?

### DISCUSSION THREAD

**Discussion by bluestorm383** [Vdauphin/HeartsAndMinds#1227] [Question and answer]
Hi there! I'm currently attempting to get this mission up to play with a few friends, and I'm attempting to limit the arsenal choices to a specific era with selections from a few mods. I have all the classnames ready to go and I have edited the define_mod.sqf file to my liking, however there's apparently some other parameters I need to modify for that to work. 

I thought I found the parameters I needed to modify in core>fnc>arsenal>data.sqf , however, the not much in there makes sense to me. 

Here is the block of code I am referencing;
-------------------------------------
params [
    ["_box", objNull, [objNull]],
    ["_arsenalType", 0, [0]],
    ["_arsenalRestrict", 0, [0]],
    ["_arsenalData", [], [[]]]
];
_arsenalData params [["_weapons", [], [[]]], ["_magazines", [], [[]]], ["_items", [], [[]]], ["_backpacks", [], [[]]]];

//BIS Arsenal
if (_arsenalType < 3) then {
    if (_arsenalRestrict isEqualTo 1) then {
        //add
        [_box, _weapons, false, false] call BIS_fnc_addVirtualWeaponCargo;
        [_box, _magazines, false, false] call BIS_fnc_addVirtualMagazineCargo;
        [_box, _items, false, false] call BIS_fnc_addVirtualItemCargo;
        [_box, _backpacks, false, false] call BIS_fnc_addVirtualBackpackCargo;
    };
-----------------------------------

What  do I need to modify to be able to utilize the whitelisted arsenal in-game?

Thank you for your time!

**Comment by Vdauphin** [comment]
Hello,

I think BIS Arsenal is broken, use ACE Arsenal

Also make sure you select the correct mission parameter: https://github.com/Vdauphin/HeartsAndMinds/blob/master/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/core/def/param.hpp#L350-L355

Cheers

**Reply by bluestorm383** [reply]
Thanks for pointing me in the right direction with the correct mission parameter, what value should I have it set to for it to be restricted to what I have defined in define_mod.sqf?

**Reply by Vdauphin** [reply]
https://github.com/Vdauphin/HeartsAndMinds/blob/aedac5cdc13d4285148db2e38313e0439412b9d9/%3DBTC%3Dco%4030_Hearts_and_Minds.Altis/define_mod.sqf#L15-L16

One of this two
For you I am guessing it is `"Full"`
